### PR TITLE
[devdoc] update example AST

### PR DESCRIPTION
**Pull request by inkydragon**
test code

```julia
macro ast_str(ex)
    ex |> Meta.parse |> Meta.show_sexpr
end

ast"a:b"
ast"a:b:c"

ast"[x for y in z, a in b]"
ast"T[x for y in z]"
ast"(a; b; c)"

ast"""
"some docs"
f(x) = x
"""
ast"11111111111111111111"
ast"0xfffffffffffffffff"
ast"111111111111111111111111111111111111111"

ast"""
if a
    b
elseif c
    d
else
    e
end
"""
```

### CODE REVIEW

File: doc/src/devdocs/ast.md [hunk 1]
```diff
@@ -53,8 +53,8 @@ call. Finally, chains of comparisons have their own special expression structure
 | `a&&b`      | `(&& a b)`                |
 | `x += 1`    | `(+= x 1)`                |
 | `a ? 1 : 2` | `(if a 1 2)`              |
-| `a:b`       | `(: a b)`                 |
-| `a:b:c`     | `(: a b c)`               |
+| `a:b`       | `(call : a b)`            |
+| `a:b:c`     | `(call : a b c)`          |
```
Comment (vtjnash) [inline doc/src/devdocs/ast.md:57]: This seems to now be already covered by the statement above the table that it is parsed as a call if it is not in the table.
```suggestion
```

File: doc/src/devdocs/ast.md [hunk 2]
```diff
@@ -108,7 +108,7 @@ Doc string syntax:
 f(x) = x
 ```
 
-parses as `(macrocall (|.| Core '@doc) (line) "some docs" (= (call f x) (block x)))`.
+parses as `(macrocall @doc (line) "some docs" (= (call f x) (block (line) x)))`.
```
Comment (vtjnash) [inline doc/src/devdocs/ast.md:111]: I think this was more correct before. Usually line nodes are implicit, and not represented in the devdocs here.
```suggestion
parses as `(macrocall (core @doc) (line) "some docs" (= (call f x) (block x)))`.
```
But `(core .)` is also internal shorthand for `(|.| Core '@doc)`, so maybe we shouldn't change this one at all
```suggestion
parses as `(macrocall (|.| Core '@doc) (line) "some docs" (= (call f x) (block x)))`.
```

**Comment by inkydragon** [comment]
```julia
julia> ast"""
       "some docs"
       f(x) = x
       """
(:macrocall, :(Core.var"@doc"), :(#= none:1 =#), "some docs", (:(=), (:call, :f, :x), (:block,
      :(#= none:2 =#),
      :x
    )))
julia> ast"`a b c`"
(:macrocall, :(Core.var"@cmd"), :(#= none:1 =#), "a b c")
```

Maybe this should also be changed to...
```diff
-| ``` `a b c` ``` | `(macrocall @cmd (line) "a b c")`   |
+| ``` `a b c` ``` | `(macrocall (|.| Core '@cmd) (line) "a b c")`   |
```

**Comment by JeffBezanson** [comment]
Those ASTs contain a `GlobalRef`, not an Expr with a `.` head, so that would be slightly misleading. I think we can leave them for now, and the rest of this looks good.

**Pull request merged**